### DISCUSSION TITLE
Feature to Read / Write arbitrary registers

### DISCUSSION THREAD

**Discussion by spiderdijon** [wlcrs/huawei_solar#687] [Ideas]
Is it possible to schedule a read / write of arbitrary registers and receive their raw output? This could be used for general debug / investigation. For example, I'm currently tracking down why setting my inverter to maximise self consumption appears to disable exporting. It only started doing this recently, so I assume I have set something which has left an errant value in a register somewhere.

**Comment by michi1984** [comment]
That is a feature i would also like to have. In my case, i would like to be able to derate the inverter to a specific power limit (i am planning to use it together with a Deye Sun 12k inverter and battery storage, which is capable of running a microgrid in case of grid failure, on three phases without any interruption. I would need to switch my 30 kW Huawei inverter to the generator input of the deye to charge the battery, but for this i need to derate the Huawei to a couple of kW max , e.g. 4-5 kW. If the battery gets full, the Huawei will be further derated by increasing the grid frequency in off grid mode).
